Adding support to json

luizperes opened the pull request:
This PR adds support to json as per https://github.com/cloudson/gitql/issues/27

I also created a flag `-json` _// not sure if it is the best name for the flag_

In this way: 
 - `git ql "select author,hash from commits"` returns the classic table as its output.
 - `git ql -json "select author,hash from commits"` returns a json as its output.

Please check the image below:

![gitsql_json](https://cloud.githubusercontent.com/assets/6173065/21217220/f7d3479c-c25f-11e6-8b0b-7bb908634d3f.png)

cloudson commented:
Awesome job @luizperes !! 
Really thanks for contribute so fast! 

I'm thinking in two points (not blockers to merge this):

* It would be important test in some way this change ? I look for my runtime_test file and it is bad :( 
* Maybe return values as a single csv file (ok, I confess, I am a csv lover) would be also useful. What do you think? From this we could change the flag -json to -output=json in another PR.

vigo commented:
Great idea! 👍

pchaganti commented:
👍

luizperes commented:
You are completely right, @cloudson, I will include that to your runtime_test!

Regarding the json output flags and the possibility to have more flags for output, I also think that it's awesome. Will fix it soon

KyleAMathews commented:
Brilliant!

luizperes commented:
Hi @cloudson , I am only concerning about the flag name `--output=json`, shouldn't it be like `--type=json` or something? The flag `-o` and `--output=<var>` are often used to indicate the output filename in other CLI. What do you think about `--type=<output_type>`? Do you have any other suggestion?

Thanks

cloudson commented:
Hey! 
`--type` looks nice for me!

luizperes commented:
Hi @cloudson , please check it again! I implemented the `--type` flag and implemented the test. It turns out that we will need to test the whole process of parsing, because the `json` is already the output... Thank you.

cloudson commented:
Perfect @luizperes . Is on develop. 
I will work to make a new release soon.

luizperes commented:
It sounds good! Thanks! I will also try to add new features later, such as the csv

jhirbour commented:
I'd love a way to pump data from git to MySQL/PGSQL to make some info graphics about my teams git history. CSV would work @luizperes but it might also be cool to have a SQL mode. Or a way to import from git into a RDBMS.

luizperes commented:
Thanks for the hint @jhirbour... we have already a similar discussion here https://github.com/cloudson/gitql/issues/39..
